src/components/Admin/Preview/Viewer.js

```diff
@@ -0,0 +1,100 @@
+import React, {Component} from 'react';
+import firebase from 'firebase/app';
+import 'firebase/firestore';
+import ItemSelector from '../ItemSelector';
+import ImagePreview from './ImagePreview';
+import getStore from '../../../utils/honestyStore.js';
+
+export default class Viewer extends Component {
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			item: 'all',
+			images: []
+		};
+
+		this.items = {
+			all: {name: 'All Items', id: 'all'},
+			unknown: {name: 'Unknown Item', id: 'unknown'}
+		};
+	}
+
+	componentDidMount() {
+		firebase.initializeApp({
+			apiKey: 'AIzaSyBVuVNKx-rx2ON0RxbfGfbGPpiymbMrxj8',
+			authDomain: 'honesty-store-kiosk.firebaseapp.com',
+			projectId: 'honesty-store-kiosk'
+		});
+		this.db = firebase.firestore();
+		this.db.settings({timestampsInSnapshots: true});
+
+		getStore().then(store => {
+			Object.assign(this.items, store);
+			this.forceUpdate();
+		});
+	}
+
+	getImages = () => {
+		let ref = this.db.collection('training_data');
+		if (this.state.item !== 'all')
+			ref = ref.where('label', '==', this.state.item);
+		ref.get().then(rows => {
+			let images = [];
+			rows.forEach(row => {
+				const img = row.data();
+				images.push({
+					id: image.id,
+					uri: img.img,
+					item: img.label,
+					trusted: img.trusted
```

Comment:
why is it images.id and not img.id?

Also would this be better as a map function? (using a return statement)